File: core/src/main/java/com/triangleleft/flashcards/mvp/main/IMainView.java
package com.triangleleft.flashcards.mvp.main;

import com.annimon.stream.Optional;
import com.triangleleft.flashcards.mvp.common.view.IView;
import com.triangleleft.flashcards.service.settings.Language;
import com.triangleleft.flashcards.service.vocabular.VocabularyWord;

import java.util.List;

public interface IMainView extends IView {

    void setTitle(String title);

    void showWord(Optional<VocabularyWord> word);

    void showList();

    void finish();

    void reloadList();

    void showDrawerProgress();

    void showUserData(String username, String avatar, List<Language> languages);

    void navigateToLogin();

    void showDrawerError();

    enum Page {
        LIST,
        WORD
    }
}

File: app/src/main/kotlin/com/triangleleft/flashcards/ui/main/IMainView.kt
package com.triangleleft.flashcards.ui.main

import com.triangleleft.flashcards.service.vocabular.VocabularyWord
import com.triangleleft.flashcards.ui.ViewEvent
import com.triangleleft.flashcards.ui.common.view.IView
import io.reactivex.Observable

interface IMainView : IView<MainViewState, ViewEvent> {

    fun backPresses(): Observable<Any>

    fun wordClicks(): Observable<VocabularyWord>

}
